drivers/crypto/crypto_stm32_hash.c

```diff
@@ -62,12 +62,30 @@ static int stm32_hash_handler(struct hash_ctx *ctx, struct hash_pkt *pkt, bool f
 
 	switch (session->algo) {
 	case CRYPTO_HASH_ALGO_SHA224:
-		status = HAL_HASHEx_SHA224_Start(&data->hhash, pkt->in_buf, pkt->in_len,
-						 pkt->out_buf, HAL_MAX_DELAY);
+#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
+		/* Update configuration now for this session */
+		data->hhash.Init.Algorithm = HASH_ALGOSELECTION_SHA224;
+		if (HAL_HASH_SetConfig(&data->hhash, &data->hhash.Init) != HAL_OK) {
+			LOG_ERR("Peripheral algo config error");
+			return -EIO;
+		}
+#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
+		LOG_DBG("HASH compute SHA224");
+		status = hal_func_hash_SHA224_start(&data->hhash, pkt->in_buf, pkt->in_len,
+						 pkt->out_buf);
 		break;
 	case CRYPTO_HASH_ALGO_SHA256:
-		status = HAL_HASHEx_SHA256_Start(&data->hhash, pkt->in_buf, pkt->in_len,
-						 pkt->out_buf, HAL_MAX_DELAY);
+#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
+		/* Update configuration now for this session */
+		data->hhash.Init.Algorithm = HASH_ALGOSELECTION_SHA256;
+		if (HAL_HASH_SetConfig(&data->hhash, &data->hhash.Init) != HAL_OK) {
+			LOG_ERR("Peripheral algo config error");
+			return -EIO;
+		}
+#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
+		LOG_DBG("HASH compute SHA256");
+		status = hal_func_hash_SHA256_start(&data->hhash, pkt->in_buf, pkt->in_len,
+						 pkt->out_buf);
```

Comment:
From what I understand this comes from the fact that H7R/S doesn't provide a `HAL_HASHEx_SHA256_Start` implementation.
Is that coming from a HW variation ? 
If yes a compatible would be required.
If not, would it be possible to use the same `HAL_HASH_SetConfig/HAL_HASH_Start` sequence on U5 as well to maximize code factorization ?

Reply:
I think `hal_func_hash_SHA256_start()` could implement the call to `HAL_HASH_SetConfig()` for the H7RSX case.

Reply:
> I think `hal_func_hash_SHA256_start()` could implement the call to `HAL_HASH_SetConfig()` for the H7RSX case.

That was an option but in this case, requires to retrieve  the data->hash.Init  variable  :  the hal_func_hash_SHA256_start is only passing a HASH_HandleTypeDef and the HAL_HASH_SetConfig also has a HASH_ConfigTypeDef parameter

Reply:
In other words, the stm32U5 has an HASH IP similar to the stm32WBA or stm32H7RS but HAL driver similar to stm32H7 or stm32F4.